apps/api/src/bootstrap.ts

```diff
@@ -137,3 +139,13 @@ function isWidgetRoute(url: string) {
 function isBlueprintRoute(url: string) {
   return url.startsWith('/v1/blueprints');
 }
+
+process.on('SIGTERM', () => {
+  otelSDK
+    .shutdown()
+    .then(
+      () => console.log('SDK shut down successfully'),
+      (err) => console.log('Error shutting down SDK', err)
+    )
+    .finally(() => process.exit(0));
+});
```

Comment:
Nestjs provides shutdown hooks we can utilize: https://docs.nestjs.com/fundamentals/lifecycle-events

Reply:
I am not sure it is possible, because here we clean the otelSDK resources that are initialized outside of the nest app (and before nest app was initiated) unless I am missing something.
what we could do is maybe move this code into an application-generic/tracing file so it could be reused and we won't need to call for it in every bootstrap.

Reply:
Definitely, I was wondering if we could move this code to the `initializeOtelSdk` scope. meaning we initialize the SDK and then:

```typescript
process.on('SIGTERM', () => {
  otelSDK
    .shutdown()
    .then(
      () => console.log('SDK shut down successfully'),
      (err) => console.log('Error shutting down SDK', err)
    )
    .finally(() => process.exit(0));
});
```
then return the SDK instance, this way we could have the `SIGTERM` listener inside of the `application-generic/tracing` file so we wouldn't need to add this listener separately in each bootstrap.